recipes/gtsam/all/conanfile.py

```diff
@@ -109,7 +106,7 @@ def config_options(self):
 
     def configure(self):
         self.requires("boost/1.72.0")
-        self.options["boost"].shared = not self.options.boost_use_static_libs
+        self.options["boost"].shared = False
```

Comment:
Hard coding options is frowned upon.
What I proposed was not changing `self.options["boost"].shared` but doing `self._cmake.definitions["Boost_USE_STATIC_LIBS"] = not self.options["boost"].shared`

Reply:
That makes more sense. I'll fix it in a minute.